smarts/core/road_map.py

```diff
@@ -650,6 +650,26 @@ def __eq__(self, other) -> bool:
         def _add_road(self, road: RoadMap.Road):
             raise NotImplementedError()
 
+        @property
+        def start_lane(self) -> Optional[RoadMap.Lane]:
+            "Route's start lane."
+            return None
+
+        @start_lane.setter
+        def start_lane(self,value:RoadMap.Lane):
+            "Route's start lane."
+            raise NotImplementedError()
```

Comment:
I am somewhat averse to mutability here since none of the other attributes of the Route are mutable. It means that the hash has to exclude the lanes because the hash would not remain consistent.

I think the state should only be initialized.

Reply:
As discussed, I have made changes to make the `Route` object immutable.